backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/predicate/impl/LuceneSimpleQueryStringPredicate.java

```diff
@@ -47,7 +56,54 @@ public SearchPredicate build() {
 
 		@Override
 		protected Query buildQuery() {
-			SimpleQueryParser queryParser = new SimpleQueryParser( buildAnalyzer(), buildWeights(), flags );
+			Map<String, LuceneCommonQueryStringPredicateBuilderFieldState> fieldStatesRemapped = new HashMap<>();
+			for ( LuceneCommonQueryStringPredicateBuilderFieldState state : fieldStates.values() ) {
+				fieldStatesRemapped.put( state.field().absolutePath(), state );
+			}
+			SimpleQueryParser queryParser = new SimpleQueryParser( buildAnalyzer(), buildWeights(), flags ) {
+
+				@Override
+				protected Query createFieldQuery(Analyzer analyzer, BooleanClause.Occur operator, String field,
+						String queryText, boolean quoted, int phraseSlop) {
+					var state = fieldStatesRemapped.get( field );
+					LuceneFieldCodec<?> codec = state.field().type().codec();
+					if ( codec instanceof AbstractLuceneNumericFieldCodec<?, ?> ) {
+						Number number = ( (AbstractLuceneNumericFieldCodec<?, ?>) codec ).fromString( queryText );
+						return ( (AbstractLuceneNumericFieldCodec<?, Number>) codec ).getDomain()
+								.createExactQuery( field, number );
+					}
+					return super.createFieldQuery( analyzer, operator, field, queryText, quoted, phraseSlop );
```

Comment:
I remember that you were not too excited about adding more and more things to the codec 😃 🙈 
but we need to somehow parse the string into something we can pass to the "createQuery" one way or the other... but if you have an idea ... 😃

Reply:
I wasn't, but in this case... I don't think there's a way around it, at least for parsing.

Though I'd suggest avoiding casts and using proper encapsulation, e.g. something like this:

```suggestion
					var state = fieldStatesRemapped.get( field );
					LuceneFieldCodec<?> codec = state.field().type().codec();
					return codec.createExactQuery( analyzer, operator, field, queryText, quoted, phraseSlop );
```

Or even, for better integration and error messages:

```suggestion
					var state = fieldStatesRemapped.get( field );
					LuceneFieldCodec<?> codec = state.field().type().codec();
					var builder state.field().queryElement(PredicateTypeKeys.MATCH, searchContext);
					Object parsed = codec.parse(queryText);
					builder.value(parsed, ValueConvert.NO);
					// ... some more setters ...
					return builder.build();
```

Or, even bolder, bring string parsing as a first-class feature available straight from the DSL; you'll still need some "parse" method in the codec, but it will be called indirectly by predicate builders:

```suggestion
					var state = fieldStatesRemapped.get( field );
					LuceneFieldCodec<?> codec = state.field().type().codec();
					var builder state.field().queryElement(PredicateTypeKeys.MATCH, searchContext);
					builder.value(queryText, ValueConvert.PARSE); // ValueConvert.PARSE is new, and means "parse value from string"
					// ... some more setters ...
					return builder.build();
```

Reply:
thanks! 😃

Reply:
Np.

For the parsing format, I'd suggest aligning with "Parsing method for 'indexNullAs'" in https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#mapping-directfieldmapping-supported-types.

Maybe we'll want to update that to say "Parsing method for 'indexNullAs' and query strings" instead :thinking: And link to there from the query string predicate's documentation.

Reply:
> For the parsing format, I'd suggest aligning with "Parsing method for 'indexNullAs'" in https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#mapping-directfieldmapping-supported-types.

That would, of course, require that bridges be able to push their parsing method to the backend, e.g. for enums.

Maybe some `queryStringConverter(ToDocumentValueConverter<String, ? extends F>)` method you could add alongside `org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeConverterStep#dslConverter`?

But with defaults for at least some of the field types, so that fields declared through something else than `ValueBridge` still have decent parsing support.

Reply:
See also:

https://github.com/hibernate/hibernate-search/blob/7b6b15c04f0ce6ec98e8d4983605b97f36d34fd5/mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/ValueBridge.java#L50-L60

https://github.com/hibernate/hibernate-search/blob/b476e2958180757b7de968a043bf7be6d1be005d/mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/ValueBindingContextImpl.java#L164-L168

https://github.com/hibernate/hibernate-search/blob/2fd171a5962eae87bd9821d3155217b2d54182e4/documentation/src/test/java/org/hibernate/search/documentation/mapper/orm/binding/fieldtypes/dsl/dslconverter/ISBNBinder.java#L30-L33